Like/Dislike functionality

Lund5teen opened the pull request.

Lund5teen committed:
Like and dislike properties and methods

Lund5teen committed:
Method added to shared model

Lund5teen committed:
Think migration fixed

Lund5teen committed:
A lot of database trial and error migrations and some like/dislike aditions

Lund5teen committed:
Follow button reintroduced

Lund5teen committed:
Fixed the dislike button

Lund5teen committed:
Liking/disliking now removes the alternative action

Lund5teen committed:
Like buttons added and fixed some visual stuff

Lund5teen committed:
Now updates amount color with themechange

Lund5teen committed:
User can no longer like/dislike own cheeps

Lund5teen committed:
Added like/dislike to private page

Lund5teen committed:
Fixed my page and additional bugs

Lund5teen committed:
Exteded deletion with new like and dislike funcitonality

Lund5teen committed:
Fixed some error message and redirection bugs

Lund5teen committed:
Merge branch 'main' into Like-Dislike

Teis-friberg reviewed (approved):
Seems good, The changes to author and cheeps seem reasonable, and in line with previous methods.

Skimmed over the CSHTML changes quickly as those are quite opaque, it works and nothing stands out.

The tests still run, did not try playwright and the factory-integration tests.

Ran the program and tried breaking it, the only thing that approaches a flaw is that likes stay after you delete an account. The liked/disliked by relation goes away. but the like post count remains. so you can keep creating/deleting the same account to spam up/downvotes on posts. This is not really an issue though? but it was all I found.

Teis-friberg committed:
Fixed private timeline exception.

Added a check to the "cannot like/dislike" code,
now it also triggers if the user is not logged in.

previously caused an exception by trying to match a non-logged in user with an author.

Co-authored-by: Marie <mblh@itu.dk>

Teis-friberg committed:
Fixed delete cheep bug

My fix of the author exception had the unintended side effect of making the delete cheep button visible when not logged in.

added an extra check on the delete cheep button. And everything should be good now.

Blomsgaard00 committed:
updated test to new environment

Blomsgaard00 committed:
Added Like and Dislike test
